code/modules/mining/equipment/regenerative_core.dm

```diff
@@ -91,10 +82,32 @@
 				to_chat(user, span_notice("[src] are useless on the dead."))
 				return
 			if(H != user)
-				H.visible_message("[user] forces [H] to apply [src]... Black tendrils entangle and reinforce [H.p_them()]!")
+			
+				if(!is_station_level(get_turf(user)) || is_reserved_level(get_turf(user)))
+					H.visible_message(span_notice("[user] crushes [src] against [H]'s body, causing black tendrils to encover and reinforce [H.p_them()]!"))
+				else
+					H.visible_message(span_notice("[user] holds [src] against [H]'s body, coaxing the regenerating tendrils from [src]..."))
+					balloon_alert(user, "Applying core...")
+					if(!do_mob(user, H, 2 SECONDS)) //come on teamwork bonus?
+						to_chat(user, span_warning("You are interrupted, causing [src]'s tendrils to retreat back into its form."))
+						return
+					balloon_alert(user, "Core applied!")
+					H.visible_message(span_notice("[src] explodes into a flurry of tendrils, rapidly covering and reinforcing [H]'s body."))
+			
+				H.visible_message(span_notice("[user] forces [H] to apply [src]... Black tendrils entangle and reinforce [H.p_them()]!"))
 				SSblackbox.record_feedback("nested tally", "hivelord_core", 1, list("[type]", "used", "other"))
 			else
-				to_chat(user, span_notice("You start to smear [src] on yourself. Disgusting tendrils hold you together and allow you to keep moving, but for how long?"))
+			
+				if(!is_station_level(get_turf(user)) || is_reserved_level(get_turf(user)))
+					to_chat(user, span_notice("You crush [src] within your hand. Disgusting tendrils spread across your body, hold you together and allow you to keep moving, but for how long?"))
+				else
+					to_chat(user, span_notice("You hold [src] against your body, coaxing the regenerating tendrils from [src]..."))
+					balloon_alert(user, "Applying core...")
+					if(!do_after(user, 4 SECONDS, src))
+						to_chat(user, span_warning("You are interrupted, causing [src]'s tendrils to retreat back into its form."))
+						return
+					balloon_alert(user, "Core applied!")
+					to_chat(user, span_notice("[src] explodes into a flurry of tendrils, rapidly spreading across your body. They will hold you together and allow you to keep moving, but for how long?"))
```

Comment:
Don't you think it would be better if a self-applied legion core had some external tell for anybody watching?